src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -58,184 +22,20 @@ export type SignOrExecuteProps = {
-
-        {decodedData && (
-          <ErrorBoundary fallback={<></>}>
-            <ConfirmationOrder decodedData={decodedData} toAddress={safeTx?.data.to ?? ''} />
-          </ErrorBoundary>
-        )}
-
-        {!props.isRejection && decodedData && (
-          <ErrorBoundary fallback={<div>Error parsing data</div>}>
-            {isApproval && <ApprovalEditor safeTransaction={safeTx} />}
-
-            {showTxDetails && <TxData txDetails={txDetails} imitation={false} trusted />}
-
-            <DecodedTx
-              tx={safeTx}
-              txId={props.txId}
-              decodedData={decodedData}
-              showMultisend={!props.isBatch}
-              showMethodCall={
-                props.showMethodCall && !showTxDetails && !isApproval && isGenericConfirmation(decodedData)
-              }
-            />
-          </ErrorBoundary>
-        )}
-        {!isCounterfactualSafe && !props.isRejection && <BlockaidBalanceChanges />}
-      </TxCard>
-
-      {!isCounterfactualSafe && !props.isRejection && <TxChecks />}
-
-      <TxCard>
-        <ConfirmationTitle
-          variant={willExecute ? ConfirmationTitleTypes.execute : ConfirmationTitleTypes.sign}
-          isCreation={isCreation}
-        />
-
-        {safeTxError && (
-          <ErrorMessage error={safeTxError}>
-            This transaction will most likely fail. To save gas costs, avoid confirming the transaction.
-          </ErrorMessage>
-        )}
-
-        {(canExecute || canExecuteThroughRole) && !props.onlyExecute && !isCounterfactualSafe && !isDelegate && (
-          <ExecuteCheckbox onChange={setShouldExecute} />
-        )}
-
-        <NetworkWarning />
-
-        <UnknownContractError />
-
-        <Blockaid />
-
-        {isCounterfactualSafe && !isDelegate && (
-          <CounterfactualForm {...props} safeTx={safeTx} isCreation={isCreation} onSubmit={onFormSubmit} onlyExecute />
-        )}
-        {!isCounterfactualSafe && willExecute && !isDelegate && (
-          <ExecuteForm {...props} safeTx={safeTx} isCreation={isCreation} onSubmit={onFormSubmit} />
-        )}
-        {!isCounterfactualSafe && willExecuteThroughRole && (
-          <ExecuteThroughRoleForm
-            {...props}
-            safeTx={safeTx}
-            safeTxError={safeTxError}
-            onSubmit={onRoleExecutionSubmit}
-            role={(allowingRole || mostLikelyRole)!}
-          />
-        )}
-        {!isCounterfactualSafe && !willExecute && !willExecuteThroughRole && !isDelegate && (
-          <SignForm
-            {...props}
-            safeTx={safeTx}
-            isBatchable={isBatchable}
-            isCreation={isCreation}
-            onSubmit={onFormSubmit}
-          />
-        )}
-
-        {isDelegate && <DelegateForm {...props} safeTx={safeTx} onSubmit={onFormSubmit} />}
-      </TxCard>
-    </>
+  return error ? (
+    <TxCard>
+      <ErrorBoundary error={error} componentStack="SignOrExecuteForm/index" />
+    </TxCard>
```

Comment:
This doesn't look user friendly (SignOrExecute/index is too technical) in case there is an error and clicking the "Go home" link takes the user out of their safe. Imo it would be enough if it said "Something went wrong please try again" and then the user can click the Back button

<img width="728" alt="Screenshot 2024-09-17 at 15 11 52" src="https://github.com/user-attachments/assets/39664397-e986-448e-893a-98fac13f761e">

Reply:
This SignOrExcuteForm/index text just appears if you are under the dev environment @usame-algan, if you go inside the ErrorBoundary component you will find this condition:

```jsx
 {IS_PRODUCTION ? (
          <Typography color="text.primary">
            In case the problem persists, please reach out to us via our{' '}
            <ExternalLink href={HELP_CENTER_URL}>Help Center</ExternalLink>
          </Typography>
        ) : (
          <>
            {/* Error may be undefined despite what the type says */}
            <Typography color="error">{error?.toString()}</Typography>
            <Typography color="error">{componentStack}</Typography>
          </>
        )}
```

Reply:
we just changed this view to continue show the transactions details with the error notification on the top, as we already have on prod.